src/DistributedLock.Postgres/PostgresDistributedReaderWriterLock.cs

```diff
@@ -29,6 +30,14 @@ public PostgresDistributedReaderWriterLock(PostgresAdvisoryLockKey key, IDbConne
     {
     }
 
+    /// <summary>
+    /// Constructs a lock with the given <paramref name="key"/> (effectively the lock name) and <paramref name="npgsqlDataSource"/>.
+    /// </summary>
+    public PostgresDistributedReaderWriterLock(PostgresAdvisoryLockKey key, NpgsqlDataSource npgsqlDataSource)
```

Comment:
We should also take in `Action<PostgresConnectionOptionsBuilder>? options` here like in the connection string version, since those settings are all applicable _except_ for useMultiplexing which we won't support initially.

I would expect the new `CreateInternalLock` function to look similar to the connection string version, except forcing a default of `false` for UseMultiplexing, e.g.

```
// Multiplexing is currently incompatible with DbDataSource, so default it to false
options = options is null ? o => o.UseMultiplexing(false) : o => { o.UseMultiplexing(false); options(o); };
var (keepaliveCadence, useTransaction, useMultiplexing) = PostgresConnectionOptionsBuilder.GetOptions(options);

return useMultiplexing
    ? throw new NotSupportedException(...)
    : new DedicatedConnectionOrTransactionDbDistributedLock(...);
```

Reply:
Yepp, added that in thanks!